framework/include/utils/KDTree.h

```diff
@@ -39,7 +39,8 @@ class KDTree
   using KdTreeT = nanoflann::KDTreeSingleIndexAdaptor<
       nanoflann::L2_Simple_Adaptor<Real, PointListAdaptor<Point>>,
```

Comment:
Does L2_Simple_Adaptor not need an extra argument too?

Reply:
Are you saying that @dschwen didn't ~test~ attempt to compile his branch before pushing?!

Reply:
Nah, he's not *me*.  But while I'm sure it'll compile fine against the current nanoflann, I'm worried about whether it'll also work with the new nanoflann, if there's a default type update in `L2_Simple_Adaptor` there.

Reply:
Yeah, CIVET passing here is not sufficient, someone with new Nanoflann also needs to make sure these changes work with it.

Reply:
I have a new nanoflann, and had everything successfully built when the commit was in a larger PR. Let me make sure I grabbed everything I needed.

Reply:
Looking at the nanoflann code I'm pretty sure it will compile. There will be some internal type narrowing without the template argument, so I'll definitely add it. But there is no error as there is not pointer to the AccessorType being passed around.

Reply:
Confirmed. The branch at 56727aa compiled with old _and_ new nanoflann. The branch at 0d2b80d compiles with new nanoflann (and Civet will tell us about old nanoflann in a few hours).

Reply:
Oh, in the old nanoflann it is

```
	template<class T, class DataSource, typename _DistanceType = T>
	struct L2_Simple_Adaptor
```

i.e. no `AccessorType` template argument., so I'll roll back that commit and apply it when we update the libmesh submodule. That way it at least compiles.